supplier/supplier.cc

```diff
@@ -42,21 +44,12 @@ class SupplierServiceImpl final : public Supplier::Service {
   // DB maintains a mapping ID -> vector<VendorInfo*>.
   // Stores pointers to VendorInfos, which are stored in an array
   std::unordered_map<uint32_t, vector<VendorInfo*>> vendor_db_;
```

Comment:
Why not just use a vector of VendorInfos?

Reply:
Because when a vector expands it might move the elements to a new address, which will cause segfault since I stored all the old pointers in vendor_db_